fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/GroupExpressionRef.java

```diff
@@ -181,6 +181,8 @@ private boolean insert(@Nonnull final T newValue, @Nullable final Map<PlanProper
         } finally {
             Debugger.withDebugger(debugger -> debugger.onEvent(new Debugger.InsertIntoMemoEvent(newValue, Debugger.Location.END)));
         }
+        Debugger.withDebugger(debugger -> debugger.onEvent(new Debugger.InsertIntoMemoEvent(newValue, containsInMemo ? Debugger.Location.REUSED : Debugger.Location.NEW)));
```

Comment:
All of this code that is sprinkled throughout the planner package is used to do some bookkeeping in order to pin-point problems with memoization, over-exploration, over-implementation of groups. There is a new `showStats()` method on `Debugger.getDebugger()` that can be evaluated from the IDE to bring up a stats page about the last planning cycle which can be quite helpful.